pkg/controller/storage/mem/storage_binding.go

```diff
@@ -38,35 +38,60 @@ func newMemStorageBinding() *memStorageBinding {
 }
 
 func (m *memStorageBinding) List() ([]*servicecatalog.Binding, error) {
-	ret := make([]*servicecatalog.Binding, len(m.bindings))
+	copy := make([]*servicecatalog.Binding, len(m.bindings))
 	i := 0
-	for _, val := range m.bindings {
-		ret[i] = val
+	for _, bin := range m.bindings {
+		copy[i] = &servicecatalog.Binding{}
+		if err := deepCopy(bin, copy[i]); err != nil {
+			return nil, err
+		}
 		i++
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Get(name string) (*servicecatalog.Binding, error) {
-	ret, ok := m.bindings[name]
+	binding, ok := m.bindings[name]
 	if !ok {
-		return nil, errNoSuchBinding
+		return nil, errNoSuchInstance
+	}
+	copy := &servicecatalog.Binding{}
+	if err := deepCopy(binding, &copy); err != nil {
+		return nil, err
 	}
-	return ret, nil
+	return copy, nil
 }
+
 func (m *memStorageBinding) Create(bin *servicecatalog.Binding) (*servicecatalog.Binding, error) {
 	if _, err := m.Get(bin.Name); err == nil {
 		return nil, errBindingAlreadyExists
 	}
-	m.bindings[bin.Name] = bin
-	return bin, nil
+	m.bindings[bin.Name] = &servicecatalog.Binding{}
+	if err := deepCopy(bin, m.bindings[bin.Name]); err != nil {
+		return nil, err
+	}
+	copy := &servicecatalog.Binding{}
+	if err := deepCopy(bin, copy); err != nil {
+		return nil, err
+	}
+	return copy, nil
 }
+
 func (m *memStorageBinding) Update(bin *servicecatalog.Binding) (*servicecatalog.Binding, error) {
 	if _, err := m.Get(bin.Name); err != nil {
 		return nil, errNoSuchBinding
 	}
-	m.bindings[bin.Name] = bin
-	return bin, nil
+	m.bindings[bin.Name] = &servicecatalog.Binding{}
+	if err := deepCopy(bin, m.bindings[bin.Name]); err != nil {
+		return nil, err
```

Comment:
this one is a bit tricky because if the `deepCopy` failed, it needs to revert to the old binding. I'd recommend creating a `&servicecatalog.Binding{}` on its own, `deepCopy` into it, and then set that deep-copied binding into `m.bindings[bin.Name]` iff the copy succeeded. For example:

```go
copyTarget := new(servicecatalog.Binding)
if err := deepCopy(bin, copyTarget); err != nil {
   return nil, err
}
m.bindings[bin.Name] = copyTarget
...
```